### DISCUSSION TITLE
How can I store data in a column not mapped to the class using a session using imperative?

### DISCUSSION THREAD

**Discussion by jalvespinto** [sqlalchemy/sqlalchemy#7617] [Usage Questions]
I have a `Product` class that has a single attribute called `sku` and this sku comes from a json file that contains a lot more info about the product. Is there a way to persist this json information to the products table if this json data is not an attribute of `Product` nor has any relation to it besides the fact the I get the sku from it? I was looking for a solution using a session (not raw SQL language like "INSERT json to products...") and imperative mapping. tks again guys

**Comment by zzzeek** [comment]
normally you would create a private attribute, such as "_json" , that maps the column, with imperative like this:

```py
mapper_registry.map_imperatively(MyClass, my_table, properties={
   "_json": my_table.c.json_column
})
```

if you dont want to have that there, then you'd need to use mapper events like the "after_insert" event to update the row, but this would be much less efficient from a SQL execution perspective.    

your Product class already has other private attributes on it like `_sa_instance_state` so there's no reason it can't have more.

**Reply by jalvespinto** [reply]
But I also would like to avoid loading this information into the instance because I will rarely use it. Maybe I will just create another class just to hold this info and declare a product.id foreignkey. I also won't write products into the database that often. I will do it once for a bulk of products. tks for the reply.

**Reply by zzzeek** [reply]
no problem use [deferred](https://docs.sqlalchemy.org/en/14/orm/loading_columns.html?highlight=deferred#deferred):

```py
from sqlalchemy.orm import deferred
mapper_registry.map_imperatively(MyClass, my_table, properties={
   "_json": deferred(my_table.c.json_column)
})
```

you'll never see it in a SELECT unless you access it.

**Reply by jalvespinto** [reply]
great. tks
